src/main/java/com/mparticle/kits/ButtonKit.java

```diff
@@ -73,148 +75,93 @@ public ButtonKit getInstance() {
-                return;
-            }
-            final String existingReferrer = mStorage.getInstallReferrer();
-            if (!KitUtils.isEmpty(existingReferrer)) {
-                ButtonLog.visibleFormat("Installation already attributed, ignoring new value. (Existing: %s, Ignored: %s)", existingReferrer, referrerRaw);
-                return;
-            }
-            String referrer;
-            try {
-                referrer = URLDecoder.decode(referrerRaw, "UTF-8");
-            } catch (UnsupportedEncodingException e) {
-                // Let's fallback to raw value if decoding fails.
-                Log.e(TAG, "Failed to decode referrer '" + referrerRaw + "'", e);
-                referrer = referrerRaw;
-            }
-            ButtonLog.visibleFormat("Recorded installation with referrer. (Referrer: %s, Raw: %s).", referrer, referrerRaw);
-            mStorage.setInstallReferrer(referrer);
-        } catch (Throwable caught) {
-            ButtonLog.warn(TAG, "Exception while handling install referrer intent." + intent, caught);
+    public void setInstallReferrer(Intent intent) {
+        if (intent.getData() != null && intent.getData().getQueryParameter("btn_ref") != null) {
+            logDebug("Tracking incoming intent as %s", intent.toString());
+            ButtonMerchant.trackIncomingIntent(getContext(), intent);
         }
     }
 
-    /**
-     * Get the Button referrer token which should be attached to any orders reported and attributed
-     * to the Button network.
-     *
-     * <pre>
-     * final ButtonKit button = (ButtonKit) MParticle.getInstance().getKitInstance(MParticle.ServiceProviders.BUTTON);
-     * if (button != null) {
-     * final String referrerToken = button.getReferrerToken();
-     * if (referrerToken != null) {
-     *      // Use token with your order reporting.
-     *  }
-     * }
-     * </pre>
-     *
-     * @return an opaque referrer string or null.
+    /*
+     * Overrides for the CommerceListener
      */
-    public String getReferrerToken() {
-        return mStorage.getReferrer();
+
+    @Override
+    public List<ReportingMessage> logLtvIncrease(BigDecimal bigDecimal, BigDecimal bigDecimal1,
+            String s, Map<String, String> map) {
+        return null;
     }
 
-    private List<ReportingMessage> handleIntent(final Uri data, final String action) {
-        if (data == null || !Intent.ACTION_VIEW.equals(action)) {
-            return null;
+    @Override
+    public List<ReportingMessage> logEvent(CommerceEvent event) {
+
+        if (!KitUtils.isEmpty(event.getProductAction()) &&
+                event.getProductAction().equalsIgnoreCase(Product.PURCHASE) &&
+                !event.getProducts().isEmpty()) {
+            logDebug("Beginning to track order %s", event.getTransactionAttributes().getId());
+
+            Order order = prepareOrder(event);
+            ButtonMerchant.trackOrder(getContext(), order, new UserActivityListener() {
```

Comment:
Potentially add a `getApplicationContext()` method to replace `getContext()` methods so we can ensure that we aren't causing any memory leaks